cpp/src/io/orc/writer_impl.cu

```diff
@@ -673,6 +675,8 @@ encoded_data writer::impl::encode_columns(const table_device_view &view,
   gpu::EncodeOrcColumnData(chunks, chunk_streams, stream);
   stream.synchronize();
 
+  dict_data.release();
+  dict_index.release();
```

Comment:
Hm, I don't think you need to release vectors here. The variables are going out of scope in the next line, memory should get freed then. Please remove these lines are verify the peak memory usage.
```suggestion
```